src/lib/statistics/chunk_statistics/counting_quotient_filter.cpp

```diff
@@ -0,0 +1,189 @@
+#include "counting_quotient_filter.hpp"
+
+#include <cmath>
+#include <iostream>
+#include <string>
+
+#include "utils/murmur_hash.hpp"
+#include "resolve_type.hpp"
+#include "storage/storage_manager.hpp"
+#include "storage/table.hpp"
+#include "storage/create_iterable_from_column.hpp"
+#include "types.hpp"
+
+
+
+namespace opossum {
+
+template <typename ElementType>
+CountingQuotientFilter<ElementType>::CountingQuotientFilter(uint8_t quotient_bits, uint8_t remainder_bits) {
+  DebugAssert(remainder_bits == 2 || remainder_bits == 4 || remainder_bits == 8 || remainder_bits == 16
+              || remainder_bits == 32, "Only the remainder sizes 2, 4, 8, 16, and 32 are supported");
+  DebugAssert(quotient_bits > 0, "quotient size can not be zero.");
+  DebugAssert(quotient_bits + remainder_bits <= 64, "The hash length can not exceed 64 bits.");
+
+  _quotient_bits = quotient_bits;
+  _remainder_bits = remainder_bits;
+  _number_of_slots = std::pow(2, _quotient_bits);
+  _hash_bits = _quotient_bits + _remainder_bits;
+
+if (remainder_bits == 2) {

```

Comment:
Even if you don't go with variants, please use an `enum class` for the possible values of remainder_bits, that's safer